runestone/common/runestonedirective.py

```diff
@@ -94,6 +98,10 @@ def _purge_runestone_data(app, env, docname):
         runestone_data.id_to_page.pop(id_)
     runestone_data.page_to_id.pop(docname)
 
+    # Reset the problem numbering for each page.
+    env.assesscounter = 0
```

Comment:
if the .. qnum directive explicitly sets the starting value for this chapter will this override it??

Reply:
No, this occurs before any directives on the page are processed. It's used to purge any stale data for the page (hence the name).